Dockerfile

```diff
@@ -31,25 +31,29 @@ RUN apt-get update -y \
 
 WORKDIR /home/app/src
 
+COPY --link --chown=9999:9999 playbook /home/app/src/playbook
+COPY --link --chown=9999:9999 playbook-website /home/app/src/playbook-website
+COPY --link package.json .rubocop.yml .eslintrc.json .yarnrc.yml yarn.lock .npmrc ./
+COPY --link .yarn ./.yarn
+
+FROM base AS build-yarn
 # Build Library
-COPY playbook-website/package.json playbook-website/
-COPY playbook/package.json playbook/
-COPY package.json .rubocop.yml .eslintrc.json .yarnrc.yml yarn.lock .npmrc ./
-COPY .yarn ./.yarn
-RUN --mount=type=secret,id=yarnenv,required env $(cat /run/secrets/yarnenv | xargs) yarn install
+RUN --mount=type=secret,id=yarnenv,required \
+    --mount=id=yarncache,type=cache,target=/home/app/.cache/yarn,uid=9999,gid=9999,sharing=locked \
+    env $(cat /run/secrets/yarnenv | xargs) yarn install
 RUN curl https://github.com/sass/node-sass/releases/download/v4.13.0/linux-x64-64_binding.node -o node_modules/node-sass/vendor/linux-x64-64_binding.node
 
-COPY --chown=app:app playbook /home/app/src/playbook
-
+FROM base AS build-bundle
 # Bundle website
-COPY --chown=app:app playbook-website /home/app/src/playbook-website
 RUN cd playbook-website && bundle install --frozen
 
-# Setup service
-RUN chmod +x playbook-website/services/*.sh
-RUN mkdir /etc/service/puma && ln -s /home/app/src/playbook-website/services/puma.sh /etc/service/puma/run
-
 FROM base AS prod
+COPY --link --from=build-bundle /usr/local/rvm/gems /usr/local/rvm/gems
+COPY --link --from=build-yarn /home/app/src /home/app/src
 
 RUN --mount=type=secret,id=yarnenv,required cd playbook; env $(cat /run/secrets/yarnenv | xargs) yarn release
 RUN --mount=type=secret,id=yarnenv,required cd playbook-website; env $(cat /run/secrets/yarnenv | xargs) yarn release
```

Comment:
@TeamTeaTime `playbook-website` needs the assets generated by `playbook` to works, so it's not possible to parallelize both process.